ios/Plugin/Plugin.swift

```diff
@@ -283,37 +284,55 @@ public class CapacitorGoogleMaps: CAPPlugin, GMSMapViewDelegate, GMSPanoramaView
             ])
         }
     }
-    
+
     @objc func setCamera(_ call: CAPPluginCall) {
-        
+
         let viewingAngle = call.getDouble("viewingAngle") ?? 45
         let bearing = call.getDouble("bearing") ?? 270
         let zoom = call.getFloat("zoom") ?? 1
         let latitude = call.getDouble("latitude") ?? 0
         let longitude = call.getDouble("longitude") ?? 0
-                
+
         let animate = call.getBool("animate") ?? false
-        let animationDuration = call.getDouble("animationDuration") ?? 1000
-        
+        let animationDuration = call.getDouble("animationDuration") ?? 1
+
+        let coordinates = call.getArray("coordinates", AnyObject.self)
+
         DispatchQueue.main.async {
             let camera = GMSCameraPosition(latitude: latitude, longitude: longitude, zoom: zoom, bearing: bearing, viewingAngle: viewingAngle)
 
             if animationDuration != 0 && animate {
                 /* FIXME: Use animation duration */
-                self.mapViewController.GMapView.animate(to: camera)
+                if coordinates == nil {
+                  self.mapViewController.GMapView.animate(to: camera)
+                } else {
+                  var gmsBounds = GMSCoordinateBounds()
+                  for coordinate in coordinates ?? [] {
+                      let coord = CLLocationCoordinate2D(latitude: coordinate["latitude"] as! CLLocationDegrees, longitude: coordinate["longitude"] as! CLLocationDegrees)
+                      let marker = GMSMarker(position: coord)
+                      gmsBounds = gmsBounds.includingCoordinate(marker.position)
+                  }
+                  CATransaction.begin()
+                  CATransaction.setValue(animationDuration, forKey: kCATransactionAnimationDuration)
+                  CATransaction.setCompletionBlock({
+                      print("animation complete, do whatever you want here")
```

Comment:
Maybe add ```call.resolve``` or ```call.success``` here so that user is notified of completion

Reply:
Done!